crates/rpc-types-engine/src/sidecar.rs

```diff
@@ -0,0 +1,102 @@
+use alloy_consensus::{Block, BlockHeader, Transaction, EMPTY_ROOT_HASH};
+use alloy_primitives::B256;
+use alloy_rpc_types_engine::{CancunPayloadFields, MaybeCancunPayloadFields};
+use derive_more::{Constructor, From, Into};
+
+/// Container type for all available additional `newPayload` request parameters that are not present
+/// in the [`ExecutionPayload`](alloy_rpc_types_engine::ExecutionPayload) object itself.
+///
+/// Default is equivalent to pre-cancun, payloads v1 and v2.
+#[derive(Debug, Clone, Default)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct OpExecutionPayloadSidecar {
+    /// Cancun request params introduced in `engine_newPayloadV3` that are not present in the
+    /// [`ExecutionPayload`](alloy_rpc_types_engine::ExecutionPayload).
+    cancun: MaybeCancunPayloadFields,
+    /// Storage root of `L2ToL1MessagePasser.sol`, aka l2 withdrawals root, requires state to
+    /// compute, hence root is passed in sidecar.
+    ///
+    /// <https://specs.optimism.io/protocol/isthmus/exec-engine.html#update-to-executabledata>
+    isthmus: MaybeIsthmusPayloadFields,
+}
+
+/// Fields introduced in `engine_newPayloadV4` that are not present in the
+/// [`ExecutionPayload`](alloy_rpc_types_engine::ExecutionPayload) RPC object.
+#[derive(Clone, Debug, Default, PartialEq, Eq, Hash, Constructor)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct IsthmusPayloadFields {
+    /// EIP-7685 requests.
+    pub withdrawals_root: B256,
```

Comment:
since this is just one field, that can be mocked, we don't necessarily need this type, and could use option directly, but imo having this makes it consistent and easier to document

Reply:
was wondering if it more had smthg to do with serde serialisation, hence copied `MaybePayloadFields` pattern from alloy